app/client/src/store/actions.ts

```diff
@@ -52,37 +43,48 @@ export default {
         });
     });
   },
-  async runPoppunk({ commit, state }: { commit: Commit, state: RootState }) {
+  async runPoppunk(context: ActionContext<RootState, RootState>) {
+    const { state, commit } = context;
     const jsonSketches = {} as { [key: string]: Record<string, never> };
     // join all file hashes to create a project hash
     const phash = Md5.hashStr(Object.keys(state.results.perIsolate).sort().join());
     commit('setProjectHash', phash);
     Object.keys(state.results.perIsolate).forEach((element) => {
       jsonSketches[element] = JSON.parse(state.results.perIsolate[element].sketch as string);
     });
-    await axiosJsonPost('poppunk', { projectHash: phash, sketches: jsonSketches })
-      .then(() => {
-        commit('setAnalysisStatus', { assign: 'submitted', microreact: 'submitted', network: 'submitted' });
-      });
+    const response = await api(context)
+      .withError('addError')
```

Comment:
I think this will currently show a warning in the console since you're not providing a mutation to run on success. If this is intentional, and there's no mutation to do with the data on some success responses, you might want to include an `ignoreSuccess` method on apiSerivce for these cases, as we have in [hint](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/apiService.ts#L20) (and check for that in `verifyHandlers`)